src/cantypes.hpp

```diff
@@ -32,20 +78,24 @@ struct CANsignal {
 struct CANmessage {
     // Constructor required for vs2015 to be able to use initializer_list
     CANmessage(std::uint32_t _id, const std::string& _name = "",
-        std::uint32_t _dlc = 0, const std::string& _ecu = "")
+        std::uint32_t _dlc = 0, const std::vector<std::string>& _ecus = { },
+        boost::optional<std::uint32_t> _cycleTime = boost::none,
+        boost::optional<std::string> _comment = boost::none)
         : id(_id)
         , name(_name)
         , dlc(_dlc)
-        , ecu(_ecu)
+        , ecus(_ecus)
+        , cycleTime(_cycleTime)
+        , comment(_comment)
     {
     }
 
     std::uint32_t id;
     std::string name;
     std::uint32_t dlc;
-    std::string ecu;
-    std::uint32_t updateCycle{ 0 };
-    std::string initValue{ "" };
+    std::vector<std::string> ecus;
```

Comment:
This is transmitter right? Shouldn't it be just `std::string` ?
The CanSignal is what can have multiple receivers iirc: 

```message = BO_ message_id message_name ':' message_size transmitter {signal} ;```

Reply:
You are correct that only a single transmitting ECU can be defined in the message definition itself -- but there is of course another optional line that can come later in the file to specify further transmitting ECUs for a message based on its CAN ID. In the following example, two transmitting ECUs are defined for the CAN ID with ID 500:

```
BO_TX_BU_ 500 : DRIVER,IO;
```

The original message definition only contained one transmitting ECU, as that's all the DBC format allows at the point of message definition:

```
BO_ 500 IO_DEBUG: 4 IO
```

Reply:
Updated my model for this